File: PHot100/easy/Solution9.py
class Solution(object):
    def isPalindrome(self, x):
        if(x<0):
            return False
        else:
            l = list()
            while(x>0):
                l.append(x%10)
                x //= 10
            lr = l[::-1]
            return lr == l


s1 = Solution
print(s1.isPalindrome(s1,1))
print(s1.isPalindrome(s1,111))
print(s1.isPalindrome(s1,121))
print(s1.isPalindrome(s1,0))
print(s1.isPalindrome(s1,-1))
print(s1.isPalindrome(s1,123))
print(s1.isPalindrome(s1,1123))

File: java100/Solution9.java
import org.junit.Test;
import static org.junit.Assert.*;
public class Solution9 {
    public boolean isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int temp = x;
        int digit = 0;
        while(temp>0){
            digit = 10*digit+ temp%10;
            temp /= 10;
        }
        return digit ==x;
    }
    @Test
    public void Test(){
        assertTrue(isPalindrome(0));
        assertFalse(isPalindrome(-1));
        assertFalse(isPalindrome(-121));
        assertFalse(isPalindrome(112));

    }
}

